index.js

```diff
@@ -0,0 +1,251 @@
+     * @return {Promise}                        combined callback results
+     */
+    allScm(callback) {
+        return new Promise((resolve, reject) => {
+            async.mapSeries(Object.keys(this.scms), (key, cb) => {
+                const scm = this.scms[key];
+
+                callback(scm)
+                .then((ret) => {
+                    cb(null, ret);
+                }, err => cb(err));
+            }, (err, results) => {
+                let map = {};
+
+                if (err) {
+                    reject(err);
+                }
+
+                results.forEach((result) => {
+                    map = Object.assign(map, result);
+                });
+
+                resolve(map);
+            });
+        });
+    }
+
+    _addWebhook(config) {
+        return this.chooseScm(config).then(scm => scm.addWebhook(config));
+    }
+
+    _parseUrl(config) {
+        return this.chooseScm(config).then(scm => scm.parseUrl(config));
+    }
+
+    _parseHook(headers, payload) {
+        return this.chooseWebhookScm(headers, payload).then(scm => scm.parseHook(headers, payload));
+    }
+
+    _getCheckoutCommand(config) {
+        return this.chooseScm(config).then(scm => scm.getCheckoutCommand(config));
+    }
+
+    _decorateUrl(config) {
+        return this.chooseScm(config).then(scm => scm.decorateUrl(config));
+    }
+
+    _decorateCommit(config) {
+        return this.chooseScm(config).then(scm => scm.decorateCommit(config));
+    }
+
+    _decorateAuthor(config) {
+        return this.chooseScm(config).then(scm => scm.decorateAuthor(config));
+    }
+
+    _getPermissions(config) {
+        return this.chooseScm(config).then(scm => scm.getPermissions(config));
+    }
+
+    _getCommitSha(config) {
+        return this.chooseScm(config).then(scm => scm.getCommitSha(config));
+    }
+
+    _updateCommitStatus(config) {
+        return this.chooseScm(config).then(scm => scm.updateCommitStatus(config));
+    }
+
+    _getFile(config) {
+        return this.chooseScm(config).then(scm => scm.getFile(config));
+    }
+
+    _getOpenedPRs(config) {
+        return this.chooseScm(config).then(scm => scm.getOpenedPRs(config));
+    }
+
+    _getBellConfiguration() {
+        return this.allScm(scm => scm.getBellConfiguration());
+    }
+
+    _getPrInfo(config) {
+        return this.chooseScm(config).then(scm => scm.getPrInfo(config));
+    }
+
+    stats() {
+        let result = {};
+
+        Object.keys(this.scms).forEach((key) => {
+            result = Object.assign(result, this.scms[key].stats());
+        });
+
+        return result;
+    }
+
+    _getScmContexts() {
+        return Object.keys(this.scms);
+    }
+
+    _canHandleWebhook(headers, payload) {
+        return this.chooseWebhookScm(headers, payload)
+            .then(scm => scm.canHandleWebhook(headers, payload));
+    }
+
+    getDisplayName(config) {
+        if (typeof this.scms[config.scmContext] !== 'object') {
```

Comment:
I think there is a misunderstanding. Registered all pluginss are stored in `scms`, and has selected the appropriate plugin by the `scmContext`. If the given `scmContext` is not in `scms`, then `scm[scmContext]` is "undefined" and can not call `getDisplayName`. Are we on the same page?